source/state_representation/tests/testJointState.cpp

```diff
@@ -5,61 +5,125 @@
 #include <gtest/gtest.h>
 #include <unistd.h>
 
-TEST(AddTwoState, PositiveNos) {
-  Eigen::VectorXd pos1 = Eigen::VectorXd::Random(4);
-  Eigen::VectorXd pos2 = Eigen::VectorXd::Random(4);
+TEST(ZeroInitialization, PositiveNos) {
+  StateRepresentation::JointState zero = StateRepresentation::JointState::Zero("test", 3);
+  // the joint state should not be considered empty (as it is properly initialized)
+  EXPECT_FALSE(zero.is_empty());
+  // all data should be zero
+  EXPECT_TRUE(zero.data().norm() == 0);
```

Comment:
normally you should not use `==` for floating point operation comparison, so EXPECT_NEAR would be more appropriate - but this is a special case where float 0 is actually 100% 0 so I guess it's ok.

Reply:
There is also EXPECT_FLOAT_EQ i believe

Reply:
Yes you are totally right. I still would like to enforce it here as if there is even some rounding errors then something is wrong for those. @domire8 I remember trying this once and could not make it work properly. I always end up doing `EXPECT_NEAR`